tests/models/blip_2/test_modeling_blip_2.py

```diff
@@ -869,7 +869,8 @@ def test_inference_opt(self):
         prompt = "Question: which city is this? Answer:"
         inputs = processor(images=image, text=prompt, return_tensors="pt").to(torch_device, dtype=torch.float16)
 
-        predictions = model.generate(**inputs)
+        # max_length for BLIP includes prompt length, so we need more than 20 to reach EOS here
+        predictions = model.generate(**inputs, max_length=25)
```

Comment:
Let's use the `max_new_tokens` argument, which prevents prompt length-based issues :)

Reply:
This uncovered a new problem with BLIP 🙃  Apparently the code was not working properly with `max_new_tokens`.

That was fixed in PR #29585 , so this should be merged only after #29585 is merged and `main` is rebased.